core/types/types.cc

```diff
@@ -40,7 +40,7 @@ TypePtr Types::untypedUntracked() {
 }
 
 TypePtr Types::untyped(const sorbet::core::GlobalState &gs, sorbet::core::SymbolRef blame) {
-    if (sorbet::debug_mode && blame.exists()) {
+    if ((sorbet::track_untyped_blame_mode || sorbet::debug_mode) && blame.exists()) {
         return make_type<BlamedUntyped>(blame);
     } else {
         return untypedUntracked();
```

Comment:
Out of an abundance of caution, what do you think of rewriting this a little differently:

```cpp
TypePtr Types::untyped(const sorbet::core::GlobalState &gs, sorbet::core::SymbolRef blame) {
    if constexpr (!sorbet::track_untyped_blame_mode && !sorbet::debug_mode) {
        return untypedUntracked();
    }

    if (blame.exists()) {
        return make_type<BlamedUntyped>(blame);
    } else {
        return untypedUntracked();
    }
}
```